backend/src/main/java/ai/verta/modeldb/project/ProjectDAORdbImpl.java

```diff
@@ -236,39 +219,16 @@ public Project insertProject(Project project, UserInfo userInfo)
   }
 
   private void createRoleBindingsForProject(Project project, UserInfo userInfo) {
-    Role ownerRole = roleService.getRoleByName(ModelDBConstants.ROLE_PROJECT_OWNER, null);
-    roleService.createRoleBinding(
-        ownerRole,
-        new CollaboratorUser(authService, userInfo),
+    final Optional<Long> ownerId =
+        userInfo != null ? Optional.of(Long.parseLong(userInfo.getUserId())) : Optional.empty();
+    roleService.createWorkspacePermissions(
+        project.getWorkspaceServiceId(),
+        Optional.of(project.getWorkspaceType()),
         project.getId(),
-        ModelDBServiceResourceTypes.PROJECT);
-
-    if (project.getProjectVisibility().equals(ProjectVisibility.PUBLIC)) {
-      roleService.createPublicRoleBinding(project.getId(), ModelDBServiceResourceTypes.PROJECT);
-    }
-
-    createWorkspaceRoleBinding(
-        project.getWorkspaceId(),
-        project.getWorkspaceType(),
-        project.getId(),
-        project.getProjectVisibility());
-  }
-
-  private void createWorkspaceRoleBinding(
-      String workspaceId,
-      WorkspaceType workspaceType,
-      String projectId,
-      ProjectVisibility projectVisibility) {
-    if (workspaceId != null && !workspaceId.isEmpty()) {
-      roleService.createWorkspaceRoleBinding(
-          workspaceId,
-          workspaceType,
-          projectId,
-          ModelDBConstants.ROLE_PROJECT_ADMIN,
-          ModelDBServiceResourceTypes.PROJECT,
-          projectVisibility.equals(ProjectVisibility.ORG_SCOPED_PUBLIC),
-          "_GLOBAL_SHARING");
-    }
+        ownerId,
+        ModelDBServiceResourceTypes.PROJECT,
+        CollaboratorTypeEnum.CollaboratorType.READ_WRITE,
```

Comment:
Why are we defaulting to RW? I'm not sure where this method is called. Should we replace it at the source to have more context?

Reply:
This is ONLY used in the method above (insertProject) to set the initial permissions.  I can set this value to whatever makes sense for a new project.

Reply:
I inlined this call to eliminate the extra method overhead, and now it lets UAC handle the owner ID assignment and defaults to READ_ONLY